### PR TITLE
Implement crossdock tests for tracing propagation

### PR DESCRIPTION
**Pull request by yurishkuro**

### CODE REVIEW

**Comment by yurishkuro** [comment]
@abhinav @breerly ready for review

**Comment by breerly** [comment]
@yurishkuro you are a beast 👍

File: tchannel-crossdock/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            - python
+        environment:
+            - WAIT_FOR=go,java,python
+
+            - AXIS_CLIENT=go
+            - AXIS_S1NAME=go
+            - AXIS_SAMPLED=true,false
+            - AXIS_S2NAME=go,java,python
+            - AXIS_S2ENCODING=json,thrift
+            - AXIS_S3NAME=go,java,python
```
Comment (breerly) [inline tchannel-crossdock/docker-compose.yml]: nice!!!

File: tchannel-crossdock/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    crossdock:
+        image: crossdock/crossdock
+        links:
+            - go
+            - java
+            - python
+        environment:
+            - WAIT_FOR=go,java,python
+
+            - AXIS_CLIENT=go
```
Comment (breerly) [inline tchannel-crossdock/docker-compose.yml]: Any reason this is only Go? You don't want to test client impls in other langs?

File: tchannel-crossdock/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            - BEHAVIOR_TRACE=client,s1name,sampled,s2name,s2encoding,s3name,s3encoding
+
+            - REPORT=compact
+    java:
+        build: .
+        ports:
+            - "8080-8082"
+
+    go:
+        image: tchannelhub/xdock-go:dev
```
Comment (breerly) [inline tchannel-crossdock/docker-compose.yml]: should this be at `:dev`?

File: tchannel-crossdock/docker-compose.yml (hunk 2 above)
Comment (yurishkuro) [inline tchannel-crossdock/docker-compose.yml]: I should add a readme, probably. This is a _crossdock client_, the orchestrator of the behaviors, and it's only implemented in Go. Each languages implements only the "server" part.

File: tchannel-crossdock/docker-compose.yml (hunk 3 above)
Comment (yurishkuro) [inline tchannel-crossdock/docker-compose.yml]: I think so, because tchannel-go's main branch is `dev`

Comment (breerly) [inline tchannel-crossdock/docker-compose.yml]: Oh yeah, that's right 👍

File: tchannel-crossdock/docker-compose.yml (hunk 2 above)
Comment (breerly) [inline tchannel-crossdock/docker-compose.yml]: Makes sense, we have a couple tests like this in YARPC, where the "driver" of the test is only implemented once :)

File: tchannel-crossdock/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FROM maven:3.3-jdk-8
+EXPOSE 8080-8082
```
Comment (breerly) [inline tchannel-crossdock/Dockerfile]: You need all 3 of these ports?

Comment (yurishkuro) [inline tchannel-crossdock/Dockerfile]: kept it for parity with yarpc/jaeger crossdock tests, but only 80/81 are used in practice

**Comment by breerly** [comment]
lgtm sir :shipit:

**Pull request merged**